docs/datagen/intro.md

```diff
@@ -0,0 +1,51 @@
+Data Generators
+===============
+
+Data generators are a way to programmatically generate the assets and data of your mods. It allows the definition of the contents of these files in your code and the automatic generation of these files, without worrying about the specification of the files.
+
+The data generator system is loaded by the main class `net.minecraft.data.Main`. Different arguments can be passed to it, to customize which mods' data are gathered, what existing files are considered, etc. The class responsible for data generation is `net.minecraft.data.DataGenerator`.
+
+ForgeGradle provides the `runData` task to run the data generators. The IDE-specific runs generation tasks (`genEclipseRuns`, `genIntellijRuns`, `genVSCodeRuns`) also create run configurations for running the data generator.
```

Comment:
Should be a bit clearer here - FG generates the runs that you define in your `runs` block. The MDK ships with the client/server/data setup, but it's not necessarily a given that these will be there. We need better documentation on this section in general, so a quick blurb here about it is fine.

Reply:
Might add a phrase stating that this applies to the `default configuration in the MDK build.gradle`, because someone who changes the runs in the `build.gradle` should be able to figure out what now runs the data generators for them.